subtrack.MAUI/Services/SubscriptionsCalculator.cs

```diff
@@ -58,14 +58,26 @@ public DateTime GetNextPaymentDate(Subscription subscription)
     {
         int startDay = subscription.FirstPaymentDay;
         var lastPayment = subscription.LastPayment;
-
-        var nextMonthDate = lastPayment.AddMonths(1);
-        var nextMonthTotalDays = DateTime.DaysInMonth(nextMonthDate.Year, nextMonthDate.Month);
-
-        if (startDay > nextMonthTotalDays)
-            return nextMonthDate;
-
-        return new DateTime(nextMonthDate.Year, nextMonthDate.Month, startDay);
+        switch (subscription.BillingOccurrence)
+        {
+            case BillingOccurrence.Week:
```

Comment:
Need to handle interval aswell. Some subs might be bi-weekly, some might be paid every 3 months